tda/matrix.py

```diff
@@ -0,0 +1,289 @@
+from copy import copy, deepcopy
+from itertools import chain
+
+
+class Matrix(object):
+    """
+
+    Representation of matrices.
+
+    """
+
+    def __init__(self, entries):
+        """
+
+        Parameters:
+        -----------
+        entries: list
+            Two dimensional list of entries.
+
+        Notes:
+        ------
+        Entries have to be integers.
```

Comment:
Add a condition to check if the input is a list of non-empty lists of integers. An exception should be raised if something other than lists of integers is passed.